### DISCUSSION TITLE
DecoderHints TryDownscale with FixedThreshold binarizer

### DISCUSSION THREAD

**Discussion by duanqn** [zxing-cpp/zxing-cpp#394] [Q&A] [answered]
Hi,
Is it correct that if I use FixedThreshold binarizer, then the downscaling doesn't help much so I should set TryDownscale to false?

**Comment by axxel** [comment]
I don't see how the choice of the binarizer would affect the usability of the tryDownscale feature. If you have only low-res/small symbols in your input or your input is super clean (no noise) then the downscaling is a waste of time.

**Comment by duanqn** [comment, marked as answer]
Recalling our discussion in #282, the downscale feature was introduced to work around some limitations in HybridBinarizer, right?
I believe ThresholdBinarizer does not have such limitations, so that's why I thought I should set tryDownscale to false when using ThresholdBinarizer, in order to save some time.

**Reply by axxel** [reply]
You are right! So `tryDownscale` is good for two things (forgot about one of them):
 1) deal with clean HybridBinarizer input that has a module size of > 8 pixels
 2) deal with very noisy input that has a module size > 4 pixels regardless of the binarizer

The first category is likely way more prevalent, so using `tryDownscale` with the `FixedThreshold` binarizer would most likely be of no use.

**Reply by duanqn** [reply]
Thanks. Does "module size" also apply to 1D barcodes? I found some cases where downscaling leads to better binarization results (with HybridBinarizer) on 1D barcodes too.

**Reply by duanqn** [reply]
Hmmm I can't mark your comment as the answer so I'll mark mine xD

**Reply by axxel** [reply]
With 1D codes only the noise advantage of the two is left.
